SPEC.md

```diff
@@ -51,401 +25,697 @@ Released versions of the spec are available as Git tags.
+4. If the plugin returns an error, halt execution and return the error to the caller.
+
+If all plugins return success, return success to the caller.
+
+### Checking an attachment
+The runtime may also ask every plugin to confirm that a given attachment is still functional. The runtime must use the same attachment parameters as it did for the _add_ operation.
+
+Checking is similar to add with two exceptions:
+- the previous result provided is always the final result of the _add_ operation.
+- If the network configuration defines `disableCheck`, then always return success to the caller.
+
+For every plugin defined in the `plugins` key of the network configuration,
+1. Look up the executable specified in the `type` field. If this does not exist, then this is an error.
+2. Derive request configuration from the plugin configuration, with the previous result from the initial _add_ operation.
+3. Execute the plugin binary, with `CNI_COMMAND=CHECK`. Provide parameters defined above as environment variables. Supply the derived configuration via standard in.
+4. If the plugin returns an error, halt execution and return the error to the caller.
+
+If all plugins return success, return success to the caller.
+
+### Deriving execution configuration from plugin configuration
+The network configuration format (which is a list of plugin configurations to execute) must be transformed to a format understood by the plugin (which is a single plugin configuration). This section describes that transformation.
+
+The execution configuration for a single plugin invocation is also JSON. It consists of the plugin configuration, primarily unchanged except for the specified additions and removals.
+
+The following fields must be inserted into the execution configuration by the runtime:
+- `cniVersion`: taken from the `cniVersion` field of the network configuration
+- `name`: taken from the `name` field of the network configuration
+- `args`: A JSON object, consisting of the generic arguments provided as an attachment parameter
+- `runtimeConfig`: A JSON object, consisting of the union of capabilities provided by the plugin and requested by the runtime (more detail below)
+- `prevResult`: A JSON object, consisting of the result type returned by the "previous" plugin. The meaning of "previous" is defined by the specific operation (_add_, _delete_, or _check_).
+
+The following fields must be **removed** by the runtime:
+- `capabilities`
+
+All other fields should be passed through unaltered.
+
+#### Deriving `runtimeConfig`
+
+Whereas `args` and CNI_ARGS are provided to all plugins, with no indication if they are going to be consumed, _Capability arguments_ need to be declared explicitly in configuration. The runtime, thus, can determine if a given network configuration supports a specific _capability_. Capabilities are not defined by the specification - rather, they are documented [conventions](CONVENTIONS.md).
```

Comment:
If `args` and `CNI_ARGS` gives different parameters on same argument, what is happen? Is that in scope of SPEC?

```
CNI_ARGS=argA=bar

{
    "cniVersion": "1.0.0",
    "name": "dbnet",
    "type": "bridge",
    "bridge": "cni0",
    "keyA": ["some more", "plugin specific", "configuration"],
    "args": {"argA": "foo"},
    "ipam": {
        "type": "host-local",
        "subnet": "10.1.0.0/16",
        "gateway": "10.1.0.1"
    },
    "dns": {
        "nameservers": [ "10.1.0.1" ]
    }
}
```